mesonbuild/backend/backends.py

```diff
@@ -1141,9 +1141,21 @@ def create_test_serialisation(self, tests: T.List['Test']) -> T.List[TestSeriali
                     cmd_args.extend(self.construct_target_rel_paths(a, t.workdir))
                 else:
                     raise MesonException('Bad object in test command.')
+
+            t_env = t.env
```

Comment:
this is not doing what you want it to do, you're just creating a reference to `t.env`. You need to import the `copy` module and do this:
```suggestion
            t_env = copy.deepcopy(t.env)
```

Reply:
:man_facepalming: thanks